### PR TITLE
Mklauncher work

### PR DESCRIPTION
**Pull request by the-snowwhite**
This Commit updates the mklauncher pythonscript to full python3 compatibility.
The Anddemo is tested to fully work on Debian Buster arm64 and amd64 + Ubuntu Focal (amd64)
To ensure the process name is always mklauncher 
a required install dependency of the python3-setproctitle package is added.

tested and ready to go

### CODE REVIEW

**Comment by the-snowwhite** [comment]
https://github.com/machinekit/machinekit-hal/issues/330

**Comment by the-snowwhite** [comment]
@cerna 
I guess the build check error is related to the addition of the `python3-setproctitle` package ?

File: src/machinetalk/mklauncher/mklauncher.py [hunk 1]
```diff
@@ -440,7 +444,8 @@ def _start_process(self, index):
                 stdout=subprocess.PIPE,
                 stderr=subprocess.STDOUT,
                 stdin=subprocess.PIPE,
-                preexec_fn=os.setsid,
+#                preexec_fn=os.setsid,
```
Comment (machinekoder) [inline src/machinetalk/mklauncher/mklauncher.py]: Please throw out the comment.

**Comment by machinekoder** [comment]
@the-snowwhite If you haven't already, please run `black` over the code. I know, there is no CI check yet, but the rest of the machinetalk/Python code is adhering to this as well https://pypi.org/project/black and likely the Python code is going to be auto-formatted in the future.

**Comment by cerna** [comment]
@the-snowwhite,
> I guess the build check error is related to the addition of the python3-setproctitle package ?

Unfortunately the build error is caused by CI/CD recipe becoming obsolete quicker than I anticipated - GitHub deprecated some calls which are currently in use in Machinekit-HAL's workflow.

They do have an `opt-in` to unsecure functionality, which I am going to try in the meantime until I can repair it correctly.

File: src/machinetalk/mklauncher/mklauncher.py (hunk 1 above)
Comment (the-snowwhite) [reply src/machinetalk/mklauncher/mklauncher.py]: OK done

**Comment by the-snowwhite** [comment]
@machinekoder 
> please run `black` over the code.

Yes I did that and it did some mostly seemingly cosmetic changes, [added that commit](https://github.com/machinekit/machinekit-hal/pull/331/commits/5c15a7c97811fbe692ce6c216f742a1eed1f1171)

**Comment by cerna** [comment]
@the-snowwhite,
rebasing on top of #332 should make the Github Actions workflow run fine.

**Comment by the-snowwhite** [comment]
I found out that the install paths for the `machinetalk/protobuf` and `machinetalk/nanopb` headers were wrong
So I added a fix for this

**Pull request closed without merging**